staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -156,7 +156,12 @@ func (s *store) Get(ctx context.Context, key string, opts storage.GetOptions, ou
 		return storage.NewInternalError(err.Error())
 	}
 
-	return decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	err = decode(s.codec, s.versioner, data, out, kv.ModRevision)
+	if err != nil {
+		metrics.RecordDecodeError(s.groupResourceString)
```

Comment:
This would need to go in GetList too, at the very least

Reply:
Added. Thank you!

Reply:
I think the correct place to capture this metric is inside `appendListItem` function, right after `codec.Decode` call because `appendListItem` does some other work after decoding.

Reply:
MSTM. Will relocate inside of `appendListItem`